File: common/src/main/java/juuxel/adorn/client/gui/screen/GameRuleDefaultsScreen.java
package juuxel.adorn.client.gui.screen;

import juuxel.adorn.client.gui.widget.Panel;
import juuxel.adorn.config.ConfigManager;
import juuxel.adorn.util.PropertyRef;
import net.minecraft.client.gui.screen.Screen;
import net.minecraft.client.gui.widget.ButtonWidget;
import net.minecraft.screen.ScreenTexts;
import net.minecraft.text.Text;

public final class GameRuleDefaultsScreen extends AbstractConfigScreen {
    private static final Layout LAYOUT = new Layout.Tabular(250, 44);

    public GameRuleDefaultsScreen(Screen parent) {
        super(Text.translatable("gui.adorn.config.game_rule_defaults"), parent, LAYOUT);
    }

    @Override
    protected void init() {
        super.init();
        addDrawableChild(
            ButtonWidget.builder(ScreenTexts.BACK, button -> close())
                .position(width / 2 - 100, height - headerFooterOptions.footerWidgetYFromBottom(BUTTON_HEIGHT))
                .size(200, 20)
                .build()
        );
    }

    @Override
    protected void initConfigWidgets(Panel panel) {
        var config = ConfigManager.config();
        addConfigToggle(PropertyRef.ofField(config.gameRuleDefaults, "skipNightOnSofas"));
        addConfigToggle(PropertyRef.ofField(config.gameRuleDefaults, "infiniteKitchenSinks"));
        addConfigToggle(PropertyRef.ofField(config.gameRuleDefaults, "dropLockedTradingStations"));
    }

    @Override
    protected String getOptionTranslationKey(String name) {
        return "gamerule.adorn:" + name;
    }
}

File: common/src/main/kotlin/juuxel/adorn/client/gui/screen/GameRuleDefaultsScreen.kt
package juuxel.adorn.client.gui.screen

import juuxel.adorn.config.ConfigManager
import juuxel.adorn.util.property
import net.minecraft.client.gui.screen.Screen
import net.minecraft.client.gui.widget.ButtonWidget
import net.minecraft.screen.ScreenTexts
import net.minecraft.text.Text

class GameRuleDefaultsScreen(parent: Screen) : AbstractConfigScreen(Text.translatable("gui.adorn.config.game_rule_defaults"), parent) {
    override fun init() {
        super.init()
        val config = ConfigManager.config()
        addConfigToggle(BUTTON_WIDTH, config.gameRuleDefaults.property { it::skipNightOnSofas })
        addConfigToggle(BUTTON_WIDTH, config.gameRuleDefaults.property { it::infiniteKitchenSinks })
        addConfigToggle(BUTTON_WIDTH, config.gameRuleDefaults.property { it::dropLockedTradingStations })
        addDrawableChild(
            ButtonWidget.builder(ScreenTexts.BACK) { close() }
                .position(this.width / 2 - 100, this.height - 27)
                .size(200, 20)
                .build()
        )
    }

    override fun getOptionTranslationKey(name: String): String =
        "gamerule.adorn:$name"

    companion object {
        private const val BUTTON_WIDTH = 250
    }
}
